File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.java
class Solution {
    public int findMin(int[] nums) {
        int result = nums[0];

        int left = 0;
        int right = nums.length - 1;

        while (left <= right) {

            //If we're ever at a sorted subarray
            if (nums[left] < nums[right]) {
                result = Math.min(result, nums[left]);
                break;
            }

            int mid = (left + right) / 2;
            result = Math.min(result, nums[mid]);

            if (nums[mid] >= nums[left]) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return result;
    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-2.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        result = [1] * len(nums)
        
        prefix = 1
        for x in range(len(nums)):
            result[x] = prefix
            prefix *= nums[x]

        postfix = 1
        for x in range(len(nums) - 1, -1, -1):
            result[x] *= postfix
            postfix *= nums[x]

        return result
        